pkg/target/host.go

```diff
@@ -169,21 +193,43 @@ func (h *Host) Flush(d time.Duration) {
 }
 
 // Connect connects to target host via TCP. If unsuccessful, sets conn to nil.
-func (h *Host) Connect() {
-	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", h.Name, h.Port),
-		time.Duration(h.ConnTimeoutSec)*time.Second)
+func (h *Host) Connect(attemptCount int) {
+	conn, err := h.getConnectionToHost()
 	if err != nil {
 		h.Lg.Warn("connection to host failed",
 			zap.String("host", h.Name),
 			zap.Uint16("port", h.Port))
 		h.Conn = nil
+		if attemptCount == 1 {
+			h.updateHostHealthStatus <- &HostStatus{Host: h, Status: false, sigCh: make(chan struct{})}
+		}
 
 		return
 	}
 
 	h.Conn = conn
+	h.updateHostHealthStatus <- &HostStatus{Host: h, Status: true, sigCh: make(chan struct{})}
 
 	h.Wm.Lock()
 	defer h.Wm.Unlock()
 	h.W = bufio.NewWriterSize(conn, h.bufSize)
 }
+
+func (h *Host) getConnectionToHost() (net.Conn, error) {
+	dialer := net.Dialer{
+		Timeout:   time.Duration(h.ConnTimeoutSec) * time.Second,
+		KeepAlive: time.Duration(h.KeepAliveSec) * time.Second,
+	}
+	conn, err := dialer.Dial("tcp", fmt.Sprintf("%s:%d", h.Name, h.Port))
```

Comment:
```suggestion
	conn, err := dialer.Dial("tcp", net.JoinHostPort(h.Name, h.Port))
```
This should work also with IPv6 hosts.